### PR TITLE
Exabeam enhance

### PR DESCRIPTION
**Pull request by merit-maita**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/40184

## Description
Fixed the *exabeam-delete-context-table-records* command to delete records based on keys passed as arguments and not ids.

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Exabeam/Integrations/Exabeam/Exabeam.py [hunk 1]
```diff
@@ -1692,7 +1692,12 @@ def delete_context_table_records(client: Client, args: Dict) -> Tuple[Any, Dict[
     session_id = args.get('session_id')
     records = argToList(args.get('records'))
 
-    record_updates_raw_data = client.delete_context_table_records_request(context_table_name, records, session_id)
+    records_raw_data = client.list_context_table_records_request(context_table_name, 10000, 1)
```
Comment (yaakovi) [inline Packs/Exabeam/Integrations/Exabeam/Exabeam.py:1695]: why aren't you using the `session_id` and `records` arguments?

File: Packs/Exabeam/Integrations/Exabeam/Exabeam.py [hunk 2]
```diff
@@ -1692,7 +1692,12 @@ def delete_context_table_records(client: Client, args: Dict) -> Tuple[Any, Dict[
...
+    all_records = records_raw_data.get('records', [])
+    filtered = list(filter(lambda record: (record['key'] in records), all_records))
+    ids = [record['id'] for record in filtered]
```
Comment (yaakovi) [inline Packs/Exabeam/Integrations/Exabeam/Exabeam.py:1698]: this is more readable and has a better performance.
```suggestion
    ids = [record['id'] for record in filtered if record['key'] in records]
```

**Review by yaakovi — CHANGES_REQUESTED** [review]
add a small unit test to verify your change

File: Packs/Exabeam/Integrations/Exabeam/Exabeam.py (hunk 1 above)
Comment (merit-maita) [reply Packs/Exabeam/Integrations/Exabeam/Exabeam.py:1695]: they're not used in the list_context_records_request function is the first place, they're used in the delete one

**Pull request merged**